src/Components /ArticleList.jsx

```diff
@@ -9,16 +9,28 @@ import { dateFormatter } from './utils';
 export default function ArticleList() {
 	const [articles, setArticles] = useState([]);
   const [isLoading, setIsLoading] = useState (true)
- 
+  const [isError, setIsError] = useState(false);
 
 useEffect(() => {
   setIsLoading(true);
-  fetchArticles().then(({ articles }) => {
-    setArticles(articles);
-    setIsLoading(false);
-  }).catch((error) => {console.log(error)})
+  fetchArticles()
+		.then(({ articles }) => {
+			setArticles(articles);
+		})
+		.catch((err) => {
+			setIsError(true);
+		})
+		.finally(() => {
+			setIsLoading(false);
+		});
+
 }, []);
 
+if (isLoading) return <p className="loading">Loading...</p>;
+
+if (isError) return <p> something went wrong, please try again later ...</p>;
+
+
 return (
   <main className="articles">
     {articles.map((article) => {
```

Comment:
Please remember to add keys to the highest level of the returned element in the map (see screenshot)
![image](https://user-images.githubusercontent.com/61753159/233325985-90da492e-43b1-4b72-8329-ea424e359acc.png)

Reply:
updated, thank you.